src/main/java/io/fabric8/maven/docker/util/ImageName.java

```diff
@@ -60,38 +63,46 @@ public ImageName(String fullName, String givenTag) {
         if (fullName == null) {
             throw new NullPointerException("Image name must not be null");
         }
-        Pattern tagPattern = Pattern.compile("^(.+?)(?::([^:/]+))?$");
-        Matcher matcher = tagPattern.matcher(fullName);
-        if (!matcher.matches()) {
-            throw new IllegalArgumentException(fullName + " is not a proper image name ([registry/][repo][:port]");
-        }
-        tag = givenTag != null ? givenTag : matcher.group(2);
-        String rest = matcher.group(1);
-
-        String[] parts = rest.split("\\s*/\\s*");
-        if (parts.length == 1) {
-            registry = null;
-            user = null;
-            repository = parts[0];
-        } else if (parts.length >= 2) {
-            if (isRegistry(parts[0])) {
-                registry = parts[0];
-                if (parts.length > 2) {
-                    user = parts[1];
-                    repository = joinTail(parts);
+
+        if(fullName.contains("@sha256")) { // Of it contains digest
+            String[] digestParts = fullName.split("@");
+            digest = digestParts[1];
+            repository = digestParts[0];
```

Comment:
What if the user specified a registry/respository@sha245:digest ? What happens with the registry ?